### PR TITLE
Various OAuth related fixes

### PR DESCRIPTION
**Pull request by matthieusieben**

### CODE REVIEW

**Comment by matthieusieben** [comment]
Fixes https://github.com/bluesky-social/atproto/issues/2816

File: packages/oauth/oauth-client/src/util.ts [hunk 1]
```diff
@@ -116,3 +117,52 @@ export class CustomEventTarget<EventDetailMap extends Record<string, unknown>> {
...
+  if (length === 1) return signals.find(Boolean)!
+
+  const controller = new AbortController()
+  const { signal } = controller
+
+  for (const signal of signals) {
+    if (signal) {
+      signal.addEventListener('abort', () => controller.abort(), { signal })
+    }
+  }
```
Comment (devinivy) [inline packages/oauth/oauth-client/src/util.ts:165]: One of the signals may also already be aborted.

File: packages/oauth/oauth-client/src/util.ts [hunk 2]
```diff
@@ -116,3 +117,52 @@ export class CustomEventTarget<EventDetailMap extends Record<string, unknown>> {
...
+  if (length === 0) return undefined
+  if (length === 1) return signals.find(Boolean)!
+
+  const controller = new AbortController()
+  const { signal } = controller
+
+  for (const signal of signals) {
+    if (signal) {
+      signal.addEventListener('abort', () => controller.abort(), { signal })
+    }
```
Comment (devinivy) [inline packages/oauth/oauth-client/src/util.ts:164]: There's shadowing going on here for `signal` which I think may cause some issues.  I assume the `{ signal }` options are intended to reference the outer `controller.signal`, to remove all listeners when `controller.abort()` occurs.

File: packages/oauth/oauth-client/src/util.ts [hunk 3]
```diff
@@ -116,3 +117,52 @@ export class CustomEventTarget<EventDetailMap extends Record<string, unknown>> {
...
+  | `${Value} ${string}`
+  | `${string} ${Value}`
+  | `${string} ${Value} ${string}`
+
+export const includesSpaceSeparatedValue = <Value extends string>(
+  input: string,
+  value: Value,
+): input is SpaceSeparatedValue<Value> => {
+  // Optimized version of:
+  // return input.split(' ').includes(value)
```
Comment (devinivy) [inline packages/oauth/oauth-client/src/util.ts:132]: Seems pretty nice!  Is there an issue with using this implementation?

File: packages/oauth/oauth-client/src/session-getter.ts [hunk 4]
```diff
@@ -205,11 +210,15 @@ export class SessionGetter extends CachedGetter<string, Session> {
   }
 
   async setStored(sub: string, session: Session) {
+    // Prevent tempering with the stored value
```
Comment (devinivy) [inline packages/oauth/oauth-client/src/session-getter.ts:213]:
```suggestion
    // Prevent tampering with the stored value
```

File: packages/oauth/oauth-client/src/util.ts (hunk 3 above)
Comment (matthieusieben) [reply packages/oauth/oauth-client/src/util.ts:132]: The `split()` implementation is [about 8 times slower](https://jsbench.me/s2m24q9yz6/1). I'm just tying to save the planet here, one CPU cycle at the time.

**Pull request merged**